### PR TITLE
Make plugin-framework provider configuration code treat empty values like the SDK

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR fixes https://github.com/hashicorp/terraform-provider-google/issues/14255

This PR makes the plugin-framework code behave more like the SDK by ignoring empty strings, and is a bug fix to be included in a v4.x.x release. The bug it's fixing is an unintentional behaviour change introduced when the provider was muxed.

Later, I will remove this fix in 5.0.0, as an intentional breaking change as part of : https://github.com/hashicorp/terraform-provider-google/issues/14447

Also, this PR changes some arbitrary strings from "123s" to "45s" as I realised that "123s" is changed to "2m3s" during the provider configuration logic. I thought this change would help avoid confusion if tests are updated in future to inspect where that second, parsed value is used.

---

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
provider: fixed the provider so it resumes ignoring empty strings set in the `provider` block
```

### CODE REVIEW

**Comment by SarahFrench** [comment]
This PR currently contains all the unit tests I've managed to get approved and merged to main, unit tests I've got in open PRs, and code changes to make the PF config code behave more like the SDK config code w.r.t empty strings

File: mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:1154]: This was a defect in the test I found when making it pass

File: mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:1066]: This is a test case that was missing, I've also added an equivalent in `.../provider/provider_internal_test.go` in this PR

**Review by SarahFrench — COMMENTED** [review]
Added some comments to help with review

File: mmv1/third_party/terraform/fwtransport/framework_config.go.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwtransport/framework_config.go.erb:67]: For 5.0.0, when we stop ignoring empty strings provided in a user's Terraform configuration, I plan to delete this section and remove the `HandleZeroValues` function

Comment (c2thorn) [reply mmv1/third_party/terraform/fwtransport/framework_config.go.erb:67]: We are adding this functionality back for users that could not upgrade since 4.60 right? To give them access to remaining 4.x versions

**Review by c2thorn — APPROVED** [review]
Straightforward except for the batching bit, but looks correct.

File: mmv1/third_party/terraform/fwtransport/framework_config.go.erb
Comment (SarahFrench) [reply mmv1/third_party/terraform/fwtransport/framework_config.go.erb:67]: Yeah that's right. Users affected by https://github.com/hashicorp/terraform-provider-google/issues/14255 could navigate the issue easily by removing empty strings from their provider config block, but that becomes less easy if they have a lot of different provider configs or if an empty string was 'baked into' a module that a user didn't have control over.

By making this change we're allowing any users on old provider versions to upgrade to the final 4.x.x version prior to 5.0.0 with everything behaving as they'd expect. Then they'll have the upgrade guide warning them about empty strings no longer being ignored in 5.0.0.

**Comment by SarahFrench** [comment]
The batching bit was a pain for me to figure out, but it works! I'll do some final manual testing tomorrow and will merge this PR then.

**Comment by SarahFrench** [comment]
Instead of manual testing, I've added an acceptance test in 16b27d4 that shows the behaviour and how it's fixed following this PR. In future I'll update that test with another step when we intentionally reverse this bug fix in 5.0.0

**Comment by SarahFrench** [comment]
Following addition of that test and all checks passing, I'll merge this PR once merging to main is resumed

**Pull request merged**
